library/LuaApi.cpp

```diff
@@ -2242,11 +2242,7 @@ int screen_show(lua_State *L)
 
     df::viewscreen *screen = dfhack_lua_viewscreen::get_pointer(L, 1, true);
 
-    bool ok = Screen::show(screen, before);
-
-    // If it is a table, get_pointer created a new object. Don't leak it.
-    if (!ok && lua_istable(L, 1))
-        delete screen;
+    bool ok = Screen::show(std::unique_ptr<df::viewscreen>{screen}, before);
```

Comment:
get_pointer has separate logic for handling tables (Screen instances from gui.lua) and raw viewscreen pointers. Does this work appropriately with both of those?
It's uncommon for Lua code to call show() directly on native screens, but it should work. I think `open-legends` and `modtools/create-unit` do this.

Reply:
Thanks. I was wondering about the native screen pointers from lua when I was looking the lua part of code. But then I forgot to find a script that uses them to test if it works. Now I checked than `open-legends` at least works correctly.